Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/activities/EmulationActivity.java

```diff
@@ -156,6 +165,14 @@ public final class EmulationActivity extends AppCompatActivity implements ThemeP
   public static final int MENU_ACTION_UNPAUSE_EMULATION = 33;
   public static final int MENU_ACTION_OVERLAY_CONTROLS = 34;
   public static final int MENU_ACTION_SETTINGS = 35;
+  public static final int MENU_ACTION_SKYLANDERS = 36;
+
+  private static Pair<Pair<Integer, Integer>, String> skylanderData;
+  private static int skylanderSlot = -1;
```

Comment:
Please prefix the names of these two variables with s if they're static... Which they shouldn't be, because you have a more important problem:

It doesn't seem like you're saving these in `onSaveInstanceState` and restoring them later. Some devices with low RAM might kill the Dolphin process while the file picker is being shown, so this is important. Note that the developer setting "Don't keep activities" is not sufficient for testing this, because it only destroys the activity, not the process.

Reply:
Think this is sorted now. Wasn't sure if I also needed to include the array as well but that hasn't been included as of yet.

Reply:
You mean `mSkylanderSlots`, right? I'm actually not entirely sure how that one is intended to work... Is it supposed to reflect the state of the Skylanders that currently have been "set" for the emulated portal? If so, how do you handle the case where the emulation activity gets created when the emulation core already has Skylanders set? I can't find any code that's fetching that information from the emulation core.

Reply:
Yeah mSkylanderSlots. The theory is that it is a representation of the Skylanders on the portal, and the array list is used to display the name of the Skylander and what slot the user has placed them in the UI. My theory was if the array list was static, the names and positions would persist when the emulation activity is created with Skylanders already placed on the portal, but it sounds like I will need to fetch the information from the core on load

Reply:
In Qt, the Window has an array m_sky_slots that has the portal slot, id, var of all of the Skylanders present on the portal, which is what I tried to do with mSkylanderSlots

Reply:
If `mSkylanderSlots` was static, I believe it would work as long as you don't use any savestates (or is savestating the currently selected Skylanders not implemented), but currently it isn't static.

Reply:
Yeah, it seems like DolphinQt has the same problem. That's not great. But since it's a bit of a bigger task and DolphinQt is broken too, I wonder if it would make sense to improve this in another PR instead...? @t895, what do you think?